libbeat/publisher/queue/diskqueue/serialize.go

```diff
@@ -82,7 +83,16 @@ func (e *eventEncoder) reset() {
 	e.folder = folder
 }
 
-func (e *eventEncoder) encode(event *publisher.Event) ([]byte, error) {
+func (e *eventEncoder) encode(evt interface{}) ([]byte, error) {
+	event, ok := evt.(publisher.Event)
+	if !ok {
+		// In order to support events of varying type, the disk queue needs
+		// to know how to encode them. When we decide to do this, we'll need
+		// to add an encoder to the settings passed in when creating a disk
+		// queue. For now, just return an error.
+		return nil, fmt.Errorf("disk queue only supports publisher.Event")
```

Comment:
This should probably be an issue, could you add one for supporting the new event type to the shipper repository?

Reply:
Thanks! https://github.com/elastic/elastic-agent-shipper/issues/41